test/attr/attr_dynamic_member_lookup.swift

```diff
@@ -688,3 +688,35 @@ func invalid_refs_through_dynamic_lookup() {
     _ = lens.baz("hello")  // expected-error {{dynamic key path member lookup cannot refer to static method 'baz'}}
   }
 }
+
+// SR-10597
+
+final class SR10597 {
+}
+
+@dynamicMemberLookup
+struct SR10597_W<T> {
+  var obj: T
+  init(_ obj: T) { self.obj = obj }
+  subscript<U>(dynamicMember member: KeyPath<T, U>) -> U {
+    return obj[keyPath: member]
+  }
+  var wooo: SR10597 { SR10597() } // expected-note {{declared here}}
+}
+
+_ = SR10597_W<SR10597>(SR10597()).wooooo // expected-error {{value of type 'SR10597_W<SR10597>' has no dynamic member 'wooooo' using key path from root type 'SR10597'; did you mean 'wooo'?}}
```

Comment:
Let's add a couple of cases here e.g. unrelated name and/or subscript to cover more use cases.